src/utils/file-handlers/recent-items-handler.js

```diff
@@ -0,0 +1,281 @@
+            resultSet.content.interface.availableLabels = true
+            resultSet.content.interface.regionTypesAllowed = true
+          } else if (
+            objectToCheck.content.interface !== objectOfRef.content.interface
+          ) {
+            //Check if the type doesn't exist or have change
+            if (!isEmpty(objectToCheck.content.interface.type)) {
+              if (isEmpty(objectOfRef.content.interface.type)) {
+                resultSet.content.interface.type = true
+              } else if (
+                objectToCheck.content.interface.type !==
+                objectOfRef.content.interface.type
+              ) {
+                resultSet.content.interface.type = true
+              }
+            }
+            //Check if the availableLabels doesn't exist or have change
+            if (!isEmpty(objectToCheck.content.interface.availableLabels)) {
+              if (isEmpty(objectOfRef.content.interface.availableLabels)) {
+                resultSet.content.interface.availableLabels = true
+              } else if (
+                objectToCheck.content.interface.availableLabels !==
+                objectOfRef.content.interface.availableLabels
+              ) {
+                resultSet.content.interface.availableLabels = true
+              }
+            }
+            //Check if the regionsTypesAllowed doesn't exist or have change
+            if (!isEmpty(objectToCheck.content.interface.regionTypesAllowed)) {
+              if (isEmpty(objectOfRef.content.interface.regionTypesAllowed)) {
+                resultSet.content.interface.regionTypesAllowed = true
+              } else if (
+                objectToCheck.content.interface.regionTypesAllowed !==
+                objectOfRef.content.interface.regionTypesAllowed
+              ) {
+                resultSet.content.interface.regionTypesAllowed = true
+              }
+            }
+          }
+        }
+        //Check if the samples doesn't exist or have change
+        if (!isEmpty(objectToCheck.content.samples)) {
+          if (isEmpty(objectOfRef.content.samples)) {
+            resultSet.content.samples = true
+          } else if (
+            objectToCheck.content.samples !== objectOfRef.content.samples
+          ) {
+            resultSet.content.samples = true
+          }
+        }
+      }
+    }
+
+    // Check if the mode doesn't exist or have change
+    if (!isEmpty(objectToCheck.mode)) {
+      if (isEmpty(objectOfRef.mode)) resultSet.mode = true
+      else if (objectToCheck.mode !== objectOfRef.mode) resultSet.mode = true
+    }
+
+    // Check if the fileName doesn't exist or have change
+    if (!isEmpty(objectToCheck.fileName)) {
+      if (isEmpty(objectOfRef.fileName)) resultSet.fileName = true
+      else if (objectToCheck.fileName !== objectOfRef.fileName)
+        resultSet.fileName = true
+    }
+    //Default behavior return false
+    return resultSet
+  }
+}
+
+export default LocalStorageHandler
```

Comment:
this code is really subject to breakage, this approach for diffing is going to be really problematic as we introduce more formats. I'm trying to understand why you need to diff the entire object manually.

* The word "handler" should probably be "helper", because "handler" may imply state. This is not a problem.

Also there's no need for a static class pattern, the pattern that would be more effective is the...

```javascript
import * as recentItemHandler from "../../recent-items-helper"
```

You would then export all the static class methods as regular methods.

Reply:
The name of this file is also confusing, because it relates to operations on the udt file format, not `recentItems`

`udt-helper.js` and in the `utils` directory would be more appropriate